controllers/front/listing/SupplierController.php

```diff
@@ -44,6 +44,15 @@ public function canonicalRedirection($canonicalURL = '')
         }
     }
 
+    public function getCanonicalURL(): string
+    {
+        if (Validate::isLoadedObject($this->supplier)) {
+            return parent::buildPaginatedUrl($this->context->link->getSupplierLink($this->supplier));
```

Comment:
```suggestion
            return $this->buildPaginatedUrl($this->context->link->getSupplierLink($this->supplier));
```

Reply:
Yes. If a controller wants to override the method, it's a direct method.